js/src/features/human-resouce/components/HumanResourceCalendar.tsx

```diff
@@ -1,72 +1,162 @@
-  moment("06.01.2020", format),
-  moment("23.01.2020", format),
-  moment("24.01.2020", format),
-  moment("25.01.2020", format),
-];
+type Props = {
+  employees: EmployeeShort[];
+  absences: AbsencesMap;
+};
 
-// TODO disable default behaviour and render working days and holidays
-const verticalLineClassNamesForTime = (timeStart: number, timeEnd: number) => {
-  const currentTimeStart = moment(timeStart);
-  const currentTimeEnd = moment(timeEnd);
+export const HumanResourceCalendar: React.FC<Props> = ({
+  employees,
+  absences,
+}) => {
+  const tableRef = useRef<HTMLTableElement>(null);
 
-  let classes = [];
-  for (let holiday of holidays) {
-    if (
-      holiday.isSame(currentTimeStart, "day") &&
-      holiday.isSame(currentTimeEnd, "day")
-    ) {
-      classes.push("holiday");
-    }
+  const [{ date, daysCount }, dispatch] = useReducer(reducer, {
+    date: new Date(),
+    daysCount: 21,
+  });
+
+  const { interval, format, title } = getDateInterval(date, daysCount);
+  const months = compose(toPairs, groupBy(formatDate("MMM")))(interval);
+
+  return (
+    <div className="content human-resource-calendar">
+      <LeftOutlined
+        style={{ userSelect: "none" }}
+        onClick={() => dispatch({ type: "move", direction: "back" })}
+      />
+      <RightOutlined
+        style={{ userSelect: "none" }}
+        onClick={() => dispatch({ type: "move", direction: "forward" })}
+      />
+      <table className="timetable" ref={tableRef}>
+        <thead>
+          <tr>
+            <th className="timetable-title" rowSpan={2}>
+              {title}
+            </th>
+            {months.map(([month, days]) => (
+              <th key={month} colSpan={days.length}>
+                {month}
+              </th>
+            ))}
+          </tr>
+          <tr>
+            {interval.map((date) => (
+              <th key={date.getTime()} className="timetable-date">
+                {formatDate(format, date)}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {employees.map((employee) => (
+            <tr key={employee.userId}>
+              <td className="timetable-user">
+                <Link to={`/users/${employee.userId}`}>
+                  {employee.firstName} {employee.lastName}
+                </Link>
+              </td>
+              {interval.map((date) => (
+                <TimelineCell
+                  key={date.toString()}
+                  date={date}
+                  absences={absences[employee.userId]}
+                />
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+type CellProps = {
+  date: Date;
+  absences: Absence[];
+};
+
+export const TimelineCell: React.FC<CellProps> = ({ date, absences = [] }) => {
+  const { className, title } = generateCellInfo(date, absences);
+
+  if (!className) {
+    return <td className="timetable-cell" />;
   }
 
-  return classes;
```

Comment:
maybe we should separate this as a component? And `generateCellInfo` can be replace with variables assignment before render?